src/java/org/apache/cassandra/repair/autorepair/AutoRepairConfig.java

```diff
@@ -308,6 +307,26 @@ public ParameterizedClass getTokenRangeSplitter(RepairType repairType)
         return applyOverrides(repairType, opt -> opt.token_range_splitter);
     }
 
+    /**
+     * Set a new token range splitter, this is not meant to be used other than for testing.
+     */
+    public void setTokenRangeSplitter(RepairType repairType, ParameterizedClass tokenRangeSplitter)
```

Comment:
maybe annotate with `@VisibleForTesting` , and can be package private
```suggestion
   @VisibleForTesting
   void setTokenRangeSplitter(RepairType repairType, ParameterizedClass tokenRangeSplitter)
```

Reply:
agreed, i was toying with the idea of being able to dynamically replace the splitter, but don't think its worth supporting as it may be a bit unpredictable.